lang/cs/Org.Apache.REEF.Tests/Functional/Failure/User/ServiceConstructorExceptionTest.cs

```diff
@@ -0,0 +1,243 @@
+                // to instantiate the RootContext.
+                Assert.Equal(0, value.FailedContexts.Count);
+                Logger.Log(Level.Info, FailedEvaluatorReceived);
+            }
+
+            /// <summary>
+            /// Submits two contexts, one that fails the Evaluator and another that allows for
+            /// Context Stacking.
+            /// </summary>
+            public void OnNext(IAllocatedEvaluator value)
+            {
+                lock (_lock)
+                {
+                    if (_shouldFailOnRootContext)
+                    {
+                        // Failing config.
+                        var ctxConf = ContextConfiguration.ConfigurationModule
+                            .Set(ContextConfiguration.Identifier, Context0)
+                            .Build();
+
+                        var serviceConf = ServiceConfiguration.ConfigurationModule
+                            .Set(ServiceConfiguration.Services, GenericType<ServiceConstructorExceptionService>.Class)
+                            .Build();
+
+                        value.SubmitContextAndService(ctxConf, serviceConf);
+                        _shouldFailOnRootContext = false;
+                    }
+                    else
+                    {
+                        // Context stacking config.
+                        value.SubmitContext(
+                            ContextConfiguration.ConfigurationModule
+                                .Set(ContextConfiguration.Identifier, Context1)
+                                .Build());
+                    }
+                }
+            }
+
+            /// <summary>
+            /// Submits the failing Context config on the non-failing Context.
+            /// </summary>
+            public void OnNext(IActiveContext value)
+            {
+                lock (_lock)
+                {
+                    Logger.Log(Level.Info, ActiveContextReceived);
+                    Assert.False(_shouldFailOnRootContext);
+
+                    var ctxConf = ContextConfiguration.ConfigurationModule
+                        .Set(ContextConfiguration.Identifier, Context2)
+                        .Build();
+
+                    var serviceConf = ServiceConfiguration.ConfigurationModule
+                        .Set(ServiceConfiguration.Services, GenericType<ServiceConstructorExceptionService>.Class)
+                        .Build();
+
+                    value.SubmitContextAndService(ctxConf, serviceConf);
+                }
+            }
+
+            /// <summary>
+            /// Only Context2 is expected as the FailedContext.
+            /// Context0 should trigger a FailedEvaluator directly.
+            /// Submit a Task on to the parent of Context2 (Context1) and make sure 
+            /// that it runs to completion.
+            /// </summary>
+            public void OnNext(IFailedContext value)
+            {
+                Assert.Equal(Context2, value.Id);
```

Comment:
So submit a contextAndService on an Evalautor, if the ContextAndService  has error it will result in IFailedEvalautor.
Submit a contextAndService on a context. if the ContextAndService  has error it will result in IFailedContext, right?